pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -121,11 +132,19 @@
   @Produces(MediaType.APPLICATION_JSON)
   @Path("/tables")
   @ApiOperation(value = "Adds a table", notes = "Adds a table")
-  public SuccessResponse addTable(String tableConfigStr) {
+  public SuccessResponse addTable(String tableConfigStr, @Context HttpHeaders httpHeaders, @Context Request request) {
     // TODO introduce a table config ctor with json string.
     TableConfig tableConfig;
+    String tableName;
     try {
       tableConfig = JsonUtils.stringToObject(tableConfigStr, TableConfig.class);
+
+      // validate permission
```

Comment:
I don't think we need to authenticate whether a certain table can be added. Pinot does not support namespace for tables. If there was one, I would agree that we can auth for a certain namespace.
We just need to verify whether someone is allowed to create a table or not. 
So, CREATE annotation should suffice.

For example , during creation, an auth system may store the user/group credentials of the creator, and may decide that only the user is allowed to update the table. Anyone may be allowed to create the table, but only owners may be allowed to update their own table (or change their schema)

Reply:
When there's any `@Authentication` annotation available on a method, authFilter check if tableName parameter is provided. If it is not, the endpoint is considered an admin level endpoint (because it's not related to any specific table). Here if we just put CREATE annotation, it means only admins can create tables and that's not what we want.
I believe what you're referring to about authentication on creation and then updating is correct, but it will be handled by implementation of `AccessControl` interface.

Reply:
Got it, thanks